vsintegration/src/FSharp.LanguageService/ProjectSitesAndFiles.fs

```diff
@@ -121,26 +127,41 @@ type internal ProjectSitesAndFiles() =
                 | _ -> None)
         | None -> Seq.empty
                     
+    static let projectCache = ConcurrentDictionary<string,FSharpProjectOptions>()
+
     static let rec referencedProjectsOf (projectSite:IProjectSite, fileName, extraProjectInfo, serviceProvider:System.IServiceProvider) =
         referencedProvideProjectSites (projectSite, serviceProvider)
         |> Seq.choose (fun (p, ps) ->            
             fullOutputAssemblyPath p
             |> Option.map (fun path ->
-                path, getProjectOptionsForProjectSite (ps.GetProjectSite(), fileName, extraProjectInfo, serviceProvider))
+                path, getProjectOptionsForProjectSite (ps.GetProjectSite(), fileName, extraProjectInfo, serviceProvider, UseCache))
             )
         |> Seq.toArray
 
-    and getProjectOptionsForProjectSite(projectSite:IProjectSite, fileName, extraProjectInfo, serviceProvider) =            
-           {ProjectFileName = projectSite.ProjectFileName()
-            ProjectFileNames = projectSite.SourceFilesOnDisk()
-            OtherOptions = projectSite.CompilerFlags()
-            ReferencedProjects = referencedProjectsOf(projectSite, fileName, extraProjectInfo, serviceProvider)
-            IsIncompleteTypeCheckEnvironment = projectSite.IsIncompleteTypeCheckEnvironment
-            UseScriptResolutionRules = SourceFile.MustBeSingleFileProject fileName
-            LoadTime = projectSite.LoadTime
-            UnresolvedReferences = None
-            OriginalLoadReferences = []
-            ExtraProjectInfo=extraProjectInfo }   
+    and getProjectOptionsForProjectSite(projectSite:IProjectSite, fileName:string, extraProjectInfo, serviceProvider, cacheUsage:CacheUsage) =
```

Comment:
This will clash with my other proposed fix.  The project options are already stored in a table in LanguageService.fs, it seems to me we should just callback to there to look up the table.